src/TypeScript/Analyzer/JavaScriptAnalyzer.cs

```diff
@@ -72,47 +60,7 @@ public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
         {
             Debug.Assert(IsAnalysisSupported(detectedLanguages));
 
-            ExecuteAnalysis(path, consumer, cancellationToken).Forget(); // fire and forget
-        }
-
-        internal async Task ExecuteAnalysis(string filePath, IIssueConsumer consumer, CancellationToken cancellationToken)
```

Comment:
After creating the virtual GetTsConfig it was possible to move everything to the base class and there wasn't any need for an async method in the derived class.